core/src/main/java/org/mule/el/context/SessionVariableMapContext.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.el.context;
+
+import org.mule.api.MuleSession;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class SessionVariableMapContext extends AbstractMapContext<String, Object>
```

Comment:
Can you refactor SessionVariableMapContext and FlowVariableMapContext to reuse as much code as possible?